Maybe.Test/MaybeTests.cs

```diff
@@ -328,5 +329,46 @@ public void OrMaybe_Func_MaybeNullArgument_ShouldThrow()
             subject.Should().ThrowExactly<ArgumentNullException>();
         }
 
+        [Fact]
+        public void ImplicitConversion_FromNullObject_ShouldReturnMaybeWithoutValue()
+        {
+            Maybe<object> subject = null;
+
+            subject.Should().Be(Maybe<object>.Nothing);
+        }
+
+        [Theory]
+        [MemberData(nameof(NonNullData))]
+        public void ImplicitConversion_FromNonNullObject_ShouldReturnMaybeWithValue<T>(T value)
+        {
+            Maybe<T> subject = value;
+
+            subject.Value.Should().Be(value);
+        }
+
+        public static TheoryData<object> NonNullData()
+        {
+            return new TheoryData<object>()
+            {
+                '1',
+                1,
+                0,
+                1.0,
+                DateTime.Now,
+                new List<string>(),
+                1.ToMaybe(),
```

Comment:
We also need to test nullable structs

Reply:
I believe implicit cast from type `T?` to type `Maybe<T>`  should not be supported right now given the complications with nullables. Maybe after implementing #36.

Reply:
I agree with @erickotsuka here. This type of implicit conversion can lead to confusion. We should be explicit in those cases (at least for now).